app/controllers/api/v1/vocab_terms_controller.rb

```diff
@@ -144,8 +164,25 @@ def show
                     include: :writeable)}
     EOS
     def update
-      standard_update(@vocab_term, Api::V1::Vocabs::TermWithDistractorsAndExerciseIdsRepresenter,
-                      user: current_api_user)
+      OSU::AccessPolicy.require_action_allowed!(:update, current_api_user, @vocab_term)
+
+      represent_with_options = {
+        user_options: { user: current_api_user },
+        represent_with: Api::V1::Vocabs::TermWithDistractorsAndExerciseIdsRepresenter
+      }
+
+      @vocab_term.with_lock do
+        consume!(@vocab_term, represent_with_options.dup)
+
+        publication_group = @vocab_term.publication.publication_group
+        if @vocab_term.save && publication_group.save
+          respond_with @vocab_term,
+                       represent_with_options.merge(responder: ResponderWithPutPatchDeleteContent)
+        else
+          render_api_errors(publication_group.errors) || render_api_errors(@vocab_term.errors)
+          raise ActiveRecord::Rollback
+        end
+      end
     end
```

Comment:
Seems like the `with_lock` and save with publication pattern has been repeated a bit.  Could add a helper:
```ruby
def model_with_publication_saver(model)
  model.with_lock do
    represent_options = yield
    if model.save && model.publication.publication-group.save
      respond_with model, represent_with_options
    else
      render_api_errors(model.errors) || render_api_errors(model.publication_group.errors)
      raise ActiveRecord::Rollback
  end
end

```

Maybe it does make sense to just repeat the bits though if they're subtly different

Reply:
Heh see the functions in https://github.com/openstax/openstax_api/blob/master/lib/openstax/api/roar.rb

But seems like every time we end up having to copy them into the apps because we need something they don't support...